ts/IterableInbox.tsx

```diff
@@ -1,25 +1,58 @@
-'use strict'
-
-import React, { Component, useState } from 'react'
-import { Text, View, StyleSheet } from 'react-native'
-import IterableInboxMessageList from './IterableInboxMessageList'
-import IterableInboxEmptyState from './IterableInboxEmptyState'
-import sampleMessages from './sampleMessageData.js'
+'use strict';
+import React, { Component, useState } from 'react';
+import { Text, View, StyleSheet } from 'react-native';
+import IterableInboxMessageList from './IterableInboxMessageList';
+import IterableInboxEmptyState from './IterableInboxEmptyState';
+import IterableInboxMessageDisplay from './IterableInboxMessageDisplay';
+import sampleMessages from './sampleMessageData.js';
 
 const IterableInbox = () => {
-   const message = "Inbox";
+   const inboxTitle = "Inbox";
+   const [isDisplayMessage, setIsDisplayMessage] = useState(false);
+   const [selectedMessageIdx, setSelectedMessageIdx] = useState(0);
    const [messages, setMessages] = useState(sampleMessages);
 
+   let selectedMessage = messages[selectedMessageIdx];
+
+   function handleMessageSelect(index: number, messages: Array<any>) {
+      if(!messages[index].read) {
+         messages[index].read = true
+      }
+      setMessages([...messages]);
+      setIsDisplayMessage(true);
+      setSelectedMessageIdx(index);
+   }
+
+   function returnToInbox() {
+      setIsDisplayMessage(false);
+   }
+   
+   function displayMessage(message: {[key: string]: any}) {
```

Comment:
Also, I am not sure about the name 'displayMessage'. It appears that this message creates a 'IterableInboxMessageDisplay' element. Maybe 'createDisplayElement'? Just a suggestion.

Reply:
I agree. Renaming to createMessageCell. In a future PR, I will separate this component into the message cell and the message cell click functionality.